File: kotlin-java8/src/jmh/java/com/example/kotlinjava8/ParallelStreamBenchmark.java
package com.example.kotlinjava8;

import java.util.concurrent.TimeUnit;
import org.openjdk.jmh.annotations.Benchmark;
import org.openjdk.jmh.annotations.BenchmarkMode;
import org.openjdk.jmh.annotations.Fork;
import org.openjdk.jmh.annotations.Measurement;
import org.openjdk.jmh.annotations.Mode;
import org.openjdk.jmh.annotations.OutputTimeUnit;
import org.openjdk.jmh.annotations.Scope;
import org.openjdk.jmh.annotations.State;
import org.openjdk.jmh.annotations.Warmup;

@State(Scope.Thread)
@BenchmarkMode(Mode.AverageTime)
@OutputTimeUnit(TimeUnit.MILLISECONDS)
@Fork(value = 2, jvmArgs = {"-Xms4G", "-Xmx4G"})
@Measurement(iterations = 2)
@Warmup(iterations = 3)
public class ParallelStreamBenchmark {

    private static final long N = 10_000_000L;

    @Benchmark
    public long iterativeSum() {
        long result = 0;
        for (long i = 1L; i <= N; i++) {
            result += i;
        }
        return result;
    }
}

File: kotlin-java8/src/test/kotlin/com/example/kotlinjava8/ParallelStreamBenchmark.kt
package com.example.kotlinjava8

import org.openjdk.jmh.annotations.*
import java.util.concurrent.TimeUnit
import java.util.stream.Stream

@State(Scope.Thread)
@BenchmarkMode(Mode.AverageTime)
@OutputTimeUnit(TimeUnit.MILLISECONDS)
@Fork(value = 2, jvmArgs = ["-Xms4G", "-Xmx4G"])
@Measurement(iterations = 2)
@Warmup(iterations = 3)


class ParallelStreamBenchmark {

    val n = 10_000_000L // 1백만

    @Benchmark
    fun asd() {


    }

}